src/DynamicDataTable.jsx

```diff
@@ -129,16 +142,19 @@ class DynamicDataTable extends Component {
     }
 
     renderRow(row) {
+        const { onClick, buttons, renderCheckBoxes, dataItemManipulator } = this.props;
+
         return (
             <DataRow
                 key={row.id}
                 row={row}
-                buttons={this.props.buttons}
+                onClick={onClick}
+                buttons={buttons}
                 fields={this.getFields()}
                 checkboxIsChecked={(value) => this.checkboxIsChecked(value)}
                 checkboxChange={(e) => this.checkboxChange(e)}
-                dataItemManipulator={(field, value) => this.props.dataItemManipulator(field, value)}
-                renderCheckboxes={this.props.renderCheckboxes}
+                dataItemManipulator={(field, value) => dataItemManipulator(field, value)}
```

Comment:
Don't use anonymous functions in the render method - they cause unnecessary re-renders and can negatively impact performance.

Reply:
This was already in the library, I just deconstructed the object